cypress/integration/block_page_test.js

```diff
@@ -0,0 +1,53 @@
+const blockTypeToLabel = {
+  consumer: 'Consumer',
+  producer: 'Producer',
+  producer_consumer: 'Producer & Consumer',
+};
+
+describe('Block page tests', () => {
+  context('no access before login', () => {
+    it('redirects to home', () => {
+      cy.visit('/blocks/blockname');
+      cy.location('pathname').should('eq', '/login');
+    });
+  });
+
+  context('authenticated', () => {
+    beforeEach(() => {
+      cy.fixture('custom_block').as('customBlock');
+      cy.fixture('native_block').as('nativeBlock');
+    });
+
+    it('successfully loads Block page for a custom block', function () {
+      cy.server();
+      cy.route('GET', '/flow/v1/*/blocks/*', '@customBlock');
+      cy.login();
+      cy.visit(`/blocks/${this.customBlock.data.name}`);
+      cy.location('pathname').should('eq', `/blocks/${this.customBlock.data.name}`);
+      cy.get('h2').contains('Block Details');
+    });
+
+    it('displays correct properties for a custom block', function () {
+      cy.contains('Name').next().contains(this.customBlock.data.name);
+      cy.contains('Type').next().contains(blockTypeToLabel[this.customBlock.data.type]);
+      cy.contains('Source');
+      cy.contains('Schema');
+    });
+
+    it('successfully loads Block page for a native block', function () {
+      cy.server();
+      cy.route('GET', '/flow/v1/*/blocks/*', '@nativeBlock');
```

Comment:
instead of declaring multiple servers/routes that match every block, you could declare the server in the before/beforeEach and use the actual block name instead of the `*` wildcard

Reply:
Good point, I'll rework this piece